src/protocolsupport/protocol/packet/middle/serverbound/play/MiddleSpecate.java

```diff
@@ -21,7 +22,12 @@ public MiddleSpecate(ConnectionImpl connection) {
 	@Override
 	public RecyclableCollection<ServerBoundPacketData> toNative() {
 		ServerBoundPacketData creator = ServerBoundPacketData.create(ServerBoundPacket.PLAY_SPECTATE);
-		MiscSerializer.writeUUID(creator, connection.getVersion(), entityUUID);
+		// FIXME: Maybe we can assume PC here..? Only implemented in v_8_9r1_9r2_10_11_12r1_12r2_13
```

Comment:
better yet. This is serverbound, the function therefore is always targetted at the latest pc version. Writing PE uuid in here would just crash the connection.